inheritance/inherit.py

```diff
@@ -0,0 +1,127 @@
+class Product:
+    def __init__(self, name: str, price: int or float, quantity: int) -> None:
+        self.name = name
+        self.price = price
+        self.quantity = quantity
+
+
+class Book(Product):
+    def __init__(self, name: str, price: int, quantity: int, author: str) -> None:
+        super().__init__(name, price, quantity)
+        self.author = author
+
+    def read(self):
+        print(
+            f'The title of book: {self.name}.\nPrice: {self.price}.\nAuthor:{self.author}\nAvailable: {self.quantity}')
+
+
+class Restaurant:
+    def __init__(self, name: str, cuisine: str, menu: dict) -> None:
+        self.name = name
+        self.cuisine = cuisine
+        self.menu = menu
+
+
+class FastFood(Restaurant):
+    def __init__(self, name: str, cuisine: str, menu: dict, drive_thru: bool) -> None:
+        super().__init__(name, cuisine, menu)
+        self.drive_thru: drive_thru
+
+    def order(self, dish_name: str, dish_quantity: int):
+        if dish_name in self.menu:
+            price, quantity = self.menu[dish_name].values()
+            left_quantity = quantity - dish_quantity
+            if left_quantity >= 0:
+                self.menu[dish_name]['quantity'] = left_quantity
+                return f'Total cost: {price * dish_quantity}'
+            return f'We are sorry.The available quantity of {dish_name} is {quantity}'
+        return f'We are sorry. The product {dish_name} is not in our menu'
```

Comment:
Якщо ми припиняємо роботу коли умови не такі як потрібно, то краще зробити early return та вийти одразу, тоді "happy path" буде без додаткових відступів та не треба буде тримати в голові усі умови за яких ми можемо продовжувати:
```suggestion
        if dish_name not in self.menu:
            return f'We are sorry. The product {dish_name} is not in our menu'

        price, quantity = self.menu[dish_name].values()
        left_quantity = quantity - dish_quantity
        if left_quantity < 0:
            return f'We are sorry. The available quantity of {dish_name} is {quantity}'
            
        self.menu[dish_name]['quantity'] = left_quantity
        return f'Total cost: {price * dish_quantity}'
```